src/lang-promql/complete/hybrid.test.ts

```diff
@@ -0,0 +1,134 @@
+        { kind: ContextKind.Aggregation },
+        { kind: ContextKind.BinOpModifier },
+      ],
+    },
+    {
+      title: 'starting to autocomplete labelName in aggregate modifier',
+      expr: 'sum by ()',
+      pos: 8, // cursor is between the bracket
+      expectedContext: [{ kind: ContextKind.LabelName }],
+    },
+    {
+      title: 'continue to autocomplete labelName in aggregate modifier',
+      expr: 'sum by (myL)',
+      pos: 11, // cursor is between the bracket after the string myL
+      expectedContext: [{ kind: ContextKind.LabelName }],
+    },
+    {
+      title: 'autocomplete labelName in a list',
+      expr: 'sum by (myLabel1, myLab)',
+      pos: 23, // cursor is between the bracket after the string myLab
+      expectedContext: [{ kind: ContextKind.LabelName }],
+    },
+    {
+      title: 'autocomplete labelName associated to a metric',
+      expr: 'metric_name{}',
+      pos: 12, // cursor is between the bracket
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: 'metric_name' }],
+    },
+    {
+      title: 'autocomplete labelName that defined a metric',
+      expr: '{}',
+      pos: 1, // cursor is between the bracket
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: '' }],
+    },
+    {
+      title: 'continue to autocomplete labelName associated to a metric',
+      expr: 'metric_name{myL}',
+      pos: 15, // cursor is between the bracket after the string myL
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: 'metric_name' }],
+    },
+    {
+      title: 'continue autocomplete labelName that defined a metric',
+      expr: '{myL}',
+      pos: 4, // cursor is between the bracket after the string myL
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: '' }],
+    },
+    {
+      title: 'autocomplete the labelValue with metricName + labelName',
+      expr: 'metric_name{labelName=""}',
+      pos: 23, // cursor is between the quotes
+      expectedContext: [{ kind: ContextKind.LabelValue, metricName: 'metric_name', labelName: 'labelName' }],
+    },
+    {
+      title: 'autocomplete aggregate operation modifier',
+      expr: 'sum() b',
+      pos: 7, // cursor is between the quotes
+      expectedContext: [{ kind: ContextKind.AggregateOpModifier }],
+    },
+    {
+      title: 'autocomplete aggregate operation modifier 2',
+      expr: 'sum b',
+      pos: 5, // cursor is after 'b'
+      expectedContext: [{ kind: ContextKind.AggregateOpModifier }, { kind: ContextKind.BinOp }],
+    },
+    {
+      title: 'autocomplete binOp',
+      expr: 'metric_name unle',
+      pos: 16,
+      expectedContext: [{ kind: ContextKind.BinOp }],
+    },
+    {
+      title: 'autocomplete matchOp',
+      expr: 'go{instance=""}',
+      pos: 12, // cursor is after the 'equal'
+      expectedContext: [{ kind: ContextKind.MatchOp }],
+    },
+    {
+      title: 'autocomplete matchOp 2',
+      expr: 'metric_name{labelName!}',
```

Comment:
This isn't working for me... if I type `metric_name{labelName!<cursor>}` then it doesn't autocomplete `!=` for me. It only autocompletes those match ops if the first character is already a valid one, like `=` completing to `=` or `=~`.

Reply:
I finally found why it doesn't work. That's because the `start` position for the completion is wrong for this case.

In the previous case it was `tree.lastChild.start` and now it is `tree.start`.

I will add some unit test to cover the calculation of the `start` position. It's a bit critical step that condition if the autocompletion will work or not even if we have properly determinated the list of the autocompletion

Reply:
fixed